dspace-server-webapp/src/main/java/org/dspace/app/rest/security/StatelessAuthenticationFilter.java

```diff
@@ -58,16 +77,19 @@ protected void doFilterInternal(HttpServletRequest req,
                                     HttpServletResponse res,
                                     FilterChain chain) throws IOException, ServletException {
 
-        Authentication authentication = getAuthentication(req);
+        inErrorOnBehalfOf = false;
+        Authentication authentication = getAuthentication(req, res);
         if (authentication != null) {
             SecurityContextHolder.getContext().setAuthentication(authentication);
             restAuthenticationService.invalidateAuthenticationCookie(res);
         }
-
+        if (inErrorOnBehalfOf) {
+            return;
+        }
```

Comment:
This `isErrorOnBehalfOf` flag is really odd to me & doesn't seem to be necessary.  It seems like you should be able to refactor this code to no longer use it....ideally you should return immediately after any call to `response.sendError()` (to avoid anything else trying to add to the response).  So, maybe this method could be refactored to simply check for the header here, and call `getOnBehalfOfAuthentication`.  If that method throws an error, catch it and call `sendError()`.  Otherwise, proceed as normal.